system-contracts/contracts/ContractDeployer.sol

```diff
@@ -30,6 +32,19 @@ contract ContractDeployer is IContractDeployer, SystemContractBase {
     /// @notice What types of bytecode are allowed to be deployed on this chain.
     AllowedBytecodeTypes public allowedBytecodeTypesToDeploy;
 
+    /// @dev Bytecode mask for delegated accounts:
+    /// - Byte 0 (0x02) means the the account is processed through the EVM interpreter
```

Comment:
Note: tooling (including BE) can rely on this byte to decide if contract is EVM or not. So this approach can confuse it

Reply:
And this is kinda error-prone, see https://github.com/matter-labs/era-contracts/pull/1518#issuecomment-2970833534

Reply:
> Note: tooling (including BE) can rely on this byte to decide if contract is EVM or not. So this approach can confuse it

This is intentional, because EIP7702 delegation _is_ an EVM contract, or rather, delegated account has EVM bytecode associated with it. Basically, the check for EIP7702 delegation available to clients now is `getCode(address) == 0xEF0100<adddress>`. So, for tooling it will actually be better to assume that the contract is EVM.
It was, in fact, the reason why I added a new flag to the EVM bytecode hash rather than introducing a new one.

However, if this approach is error-prone and creates security concerns, we surely need to analyze it and introducing new bytecode hash format could be a better option.